test/new-e2e/tests/process/windows_test.go

```diff
@@ -110,26 +113,21 @@ func (s *windowsTestSuite) TestProcessCheckIO() {
 		awshost.WithAgentOptions(agentparams.WithAgentConfig(processCheckConfigStr), agentparams.WithSystemProbeConfig(systemProbeConfigStr)),
 	))
 
-	// Flush fake intake to remove payloads that won't have IO stats
-	s.Env().FakeIntake.Client().FlushServerAndResetAggregators()
-
 	assert.EventuallyWithT(t, func(collect *assert.CollectT) {
 		assertRunningChecks(collect, s.Env().Agent.Client, []string{"process", "rtprocess"}, true)
 	}, 1*time.Minute, 5*time.Second)
 
-	// s.Env().VM.Execute("Start-MpScan -ScanType FullScan")
+	err := runDiskSpd(s.T(), s.Env().RemoteHost)
+	require.NoError(s.T(), err)
 
 	var payloads []*aggregator.ProcessPayload
 	assert.EventuallyWithT(t, func(c *assert.CollectT) {
 		var err error
 		payloads, err = s.Env().FakeIntake.Client().GetProcesses()
 		assert.NoError(c, err, "failed to get process payloads from fakeintake")
 
-		// Wait for two payloads, as processes must be detected in two check runs to be returned
-		assert.GreaterOrEqual(c, len(payloads), 2, "fewer than 2 payloads returned")
+		assertProcessCollectedNew(c, payloads, true, "diskspd.exe")
```

Comment:
Moved this check into the assert eventually so we check payloads for up to 2 minutes instead of just the first 2 we get to reduce flakiness.